src/main/kotlin/com/hypto/iam/server/idp/CognitoProviderImpl.kt

```diff
@@ -145,7 +147,12 @@ class CognitoIdentityProviderImpl : IdentityProvider, KoinComponent {
             }
         } catch (e: UsernameExistsException) {
             logger.info(e) { "Error while trying to create user = ${e.message}" }
-            throw UserAlreadyExistException("Username: ${userCredentials.userName} already present in the organization")
+            try {
+                getUser(identityGroup, userCredentials.username)
+                throw UnknownError("Unknown error while trying to create user")
```

Comment:
Add a comment on why we throw error here. Also update error message to try again after sometime as this is a retriable error.

Reply:
Added